react-js/react.md

```diff
@@ -41,29 +42,108 @@ You will spend the day working through these materials.
+
+import React from 'react';
+import { ChildComponent } from '../child/ChildComponent';
+
+export const ParentComponent = () =>
+  <div className="parent-component">
+    <ChildComponent className="parent-component__child">
+      ...
+    </ChildComponent>
+  </div>
+
+// ChildComponent.js
+
+import React from 'react';
+
+export const ChildComponent = ({ className, children }) =>
+  <div className={`some-css-className ${className}`}>
+    {children}
+  </div>
+  
+```
+will render as:
+```
+<div class="parent-component">
+  <div class="some-css-className parent-component__child">
+    ...
+  </div>
+</div>
+
+```
 
 ### Common Mistakes / Misconceptions
 
-1. JSX tag names for components must match up with component names, both in spelling and case.
-2. In render functions, don't forget to return JSX.
-3. Don't forget to specify both arguments in ReactDOM.render function.
-4. Don't forget `this` when working within a React class. Examples: `this.state.thing`
-5. Don't forget that variables in JSX must be enclosed in brackets and don't forget number 4. Example: `<div>Hello {this.state.thing}</div>`
-6. The event system in React uses attributes you are likely familiar with from HTML, but they are written using camelCase. Example: `<button onClick={function() { alert('hello!') }}>I am a button</button>`
-7. The term 'ReactJS' is commonly used to clarify that it's a javascript framework; however, 'React' is its official name.
+1. In render functions, don't forget to return JSX.
+2. Don't forget that variables in JSX must be enclosed in brackets. Example: `<div>Hello {this.state.thing}</div>`
+3. The event system in React uses attributes you are likely familiar with from HTML, but they are written using camelCase. Example: `<button onClick={function() { alert('hello!') }}>I am a button</button>`
 
 ### Guided Practice
--[Importing & Exporting with React](https://medium.com/@thejasonfile/a-simple-intro-to-javascript-imports-and-exports-389dd53c3fac)
+- [Importing & Exporting with React](https://medium.com/@thejasonfile/a-simple-intro-to-javascript-imports-and-exports-389dd53c3fac)
+  - 5 minute read article.
+- [Rendering a List using map](https://codepen.io/jamesjose03/pen/jOPRwVm)
+  - Here, we are rendering an  Ordered List using React.
+  - Use ES6 arrow function
+    - Eg: `hello = () => {
+           return "Hello World!";
+          }`
+  - Use semantic HTML, like `<ol>`
+  - If the state of this react element in the virtual dom changes, React needs to quickly figure out which element changed and where in the real dom it needs to reflect that change. So, we have to set `key` attribute on the li of the list.
+    - Eg: `<li key={value}>`
+  - If the key property is not set, then the browser will give warning in the console.
 
 ### Independent Practice
--[Create-react-app](https://medium.com/in-the-weeds/learning-react-with-create-react-app-part-1-a12e1833fdc)
+- [CodePen for Hello World Example](https://codepen.io/jamesjose03/pen/gOpVeLQ)
```

Comment:
Provide a time.  how long should they do this?

Reply:
Perfect this already exists. Make this the example you walk through above and put the parent child stuff here.

Reply:
Okay

Reply:
Are you expecting them to sped two hours on this?

Reply:
Reduced it to 30 mins now.